### PR TITLE
Convert names into prometheus metrics

### PR DESCRIPTION
**Pull request by kmada97**

### CODE REVIEW

File: sources/pmwebapi.go
Comment (roclark) [inline sources/pmwebapi.go]: If you want to make this even cleaner, you can use the `+=` operator. This adds the variable you are storing the result to at the beginning of the operation. So...
```go
pcpMetrics.Name += unit + "_total"
```
is shorthand for
```go
pcpMetrics.Name = pcpMetrics.Name + unit + "_total"
```

File: sources/pmwebapi.go
Comment (roclark) [inline sources/pmwebapi.go]: Same as above, but instead
```go
pcpMetrics.Name += unit
```

**Review by roclark — COMMENTED** [review]
Looking good! Just a couple comments to make this even cleaner, but this is a very nice commit!

**Comment by joehandzik** [comment]
Remember to add your email tag to the commit!

File: sources/pmwebapi.go
Comment (joehandzik) [inline sources/pmwebapi.go:142]: We'll want to take a look here, I don't think *everything* will have a meaningful unit. This is a place where unit tests may be able to help you faster than manual testing.

File: sources/pmwebapi.go
Comment (joehandzik) [reply sources/pmwebapi.go]: Probably also want to make sure you don't run into casing issues with "COUNTER", just to be safe. I think we've handled that elsewhere in the code already, so you should have an example to go off of.

Comment (roclark) [reply sources/pmwebapi.go]: Can you explain this one a little more? Are you referring to making sure that "COUNTER" is always allcaps or something else? If the former, we can be positive that this will always parse as allcaps since he forces the uppercase value of `pcpMetrics.Type` in the comparison.

**Comment by roclark** [comment]
Looks like you are at two commits for this now. Once you merge the two (but use the second commit's message instead of the first's!), I will approve this and let @joehandzik have the final say since he still has a few other comments.

File: sources/pmwebapi.go
Comment (joehandzik) [reply sources/pmwebapi.go]: Ah, yes, nvm, I failed to read this correctly. This looks fine.

**Comment by joehandzik** [comment]
Yup, he needs to do a rebase and meld these commits together (I think using "fixup") and add his name and email address.

**Comment by kmada97** [comment]
@joehandzik Is this fine?

**Review by roclark — APPROVED** [review]

**Comment by joehandzik** [comment]
Merging, with the caveat that I'm still not sure all units will be capitalized correctly and/or expanded the way we want (us vs microseconds, for example).

**Pull request merged**
